database/sqlite/sqlite_functions.c

```diff
@@ -0,0 +1,1224 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "sqlite_functions.h"
+
+#define ENABLE_CACHE_CHARTS 1
+#define ENABLE_CACHE_DIMENSIONS 1
+
+const char *database_config[] = {
+    "PRAGMA auto_vacuum=incremental; PRAGMA synchronous=1 ; PRAGMA journal_mode=WAL; PRAGMA temp_store=MEMORY;",
+    "PRAGMA journal_size_limit=17179869184;",
+    "CREATE TABLE IF NOT EXISTS host(host_id blob PRIMARY KEY, hostname text, "
+    "registry_hostname text, update_every int, os text, timezone text, tags text);",
+    "CREATE TABLE IF NOT EXISTS chart(chart_id blob PRIMARY KEY, host_id blob, type text, id text, name text, "
+    "family text, context text, title text, unit text, plugin text, module text, priority int, update_every int, "
+    "chart_type int, memory_mode int, history_entries);",
+    "CREATE TABLE IF NOT EXISTS dimension(dim_id blob PRIMARY KEY, chart_id blob, id text, name text, "
+    "multiplier int, divisor int , algorithm int, options text);",
+    "CREATE TABLE IF NOT EXISTS chart_active(chart_id blob PRIMARY KEY, date_created int);",
+    "CREATE TABLE IF NOT EXISTS dimension_active(dim_id blob primary key, date_created int);",
+    "CREATE TABLE IF NOT EXISTS metadata_migration(filename text, file_size, date_created int);",
+    "CREATE INDEX IF NOT EXISTS ind_host on chart(host_id);",
+    "CREATE INDEX IF NOT EXISTS ind_chart on dimension(chart_id);",
+
+    "delete from chart_active;",
+    "delete from dimension_active;",
+
+    "delete from chart where chart_id not in (select chart_id from dimension);",
+    "delete from host where host_id not in (select host_id from chart);",
```

Comment:
Are these going to take a long time with a large database? Maybe we should use them as part of a consistency check operation when a problem is detected.

Reply:
These are part of the cleanup due to rotation (cleanup during startup). I don't expect those to take a long time. 

I will run a test with an agent with 2 million dimensions and 200K charts to compare how the query plan behaves on

```
delete from chart where chart_id not in (select chart_id from dimension);

vs

delete from chart where chart_id not in (select distinct(chart_id) from dimension);
```
If both take a long time (as in more than a couple of seconds) we can test doing it when a dimension is deleted with a trigger. If that is heavy, we can just record chart_ids using a delete trigger on the dimension to a new table `chart_to_check` and then test if 

`delete from chart where chart_id not in (select c.chart_id from dimension d, chart_to_check c where d.chart_id = c.chart_id);`

behaves better. 

We will need to add vacuum logic at some point especially if we have a lot of hosts that appear just once and then deleted to reclaim space (it won't be a problem if they are deleted and reappear though as the space will be reused)